src/sentry/quotas/redis.py

```diff
@@ -23,19 +23,29 @@ def get_dynamic_cluster_from_options(setting, config):
 
 
 class BasicRedisQuota(object):
-    __slots__ = ["prefix", "subscope", "limit", "window", "reason_code", "enforce"]
+    __slots__ = ["prefix", "subscope", "limit", "window", "reason_code"]
+
+    def __init__(self, prefix=None, subscope=None, limit=None, window=None, reason_code=None):
+        if limit == 0:
```

Comment:
I added one helper constructor for zero-sized quotas, but I cannot make effective use of any other kind of constructor, so it would only be for documentation purposes. The reason for this is that we have several internal APIs such as `get_key_quota` in `base.py` that do not directly deal with `BasicRedisQuota` but just return tuples. The code that consumes the tuples does not care about limited vs unlimited and just feeds them into `__init__` unconditionally. So the branches that decide whether something is limited vs unlimited cannot decide which constructor to call because they have no knowledge of `BasicRedisQuota`.

I believe there should be a follow-up refactor where we move `BasicRedisQuota` into `base.py`. This would solve some problems I already encountered while trying to use the quotas service in the relay project config endpoint, but it effectively kills the service abstraction.

I need to think about this further and would prefer to move this into a follow-up pr instead